### PR TITLE
MediatR and handlers added

### LINKED ISSUE #8: MediatR and handlers
add MediatR to the API project https://github.com/jbogard/MediatR
move application logic to handlers. Make sure that every controller action has its own query/command, handler and response

### PR DESCRIPTION
**Pull request by mila-laurel**
closes #8

### CODE REVIEW

File: CinemaCat.Api/DTO/CreateMovieModel.cs [hunk 1]
```diff
@@ -2,4 +2,4 @@
 
 namespace CinemaCat.Api.DTO;
 
-public record class CreateMovieModel(string Title, string ReleasedDate, Person Director, int Rating);
\ No newline at end of file
+public record class CreateMovieModel(string Title, string ReleasedDate, string DirectorName, int Rating);
```
Comment (rumkit) [inline CinemaCat.Api/DTO/CreateMovieModel.cs:5]: let's leave it Person with opitonal GUID
a list of TOP actors can be also added here

File: CinemaCat.Api/Controllers/MoviesController.cs [hunk 2]
```diff
@@ -1,48 +1,57 @@
...
-            ReleasedDate = DateOnly.Parse(movie.ReleasedDate),
-            Director = movie.Director,
-            TopActors = []
+            ReleasedDate = movie.ReleasedDate,
+            Director = movie.DirectorName,
+            Rating = movie.Rating
         };
-        return await moviesProvider.CreateAsync(newValue);
+        var response = await mediator.Send(req);
+        return response.ToResult();
```
Comment (rumkit) [inline CinemaCat.Api/Controllers/MoviesController.cs:37]: let's have Created here with a link to the movie

File: CinemaCat.Api/Controllers/PersonsController.cs [hunk 3]
```diff
@@ -1,47 +1,56 @@
...
-            Person = person.Name,
-            DateOfBirth = DateOnly.Parse(person.DateOfBirth),
+            Name = person.Name,
+            DateOfBirth = person.DateOfBirth,
             PlaceOfBirth = person.PlaceOfBirth
         };
-
-        return await personsProvider.CreateAsync(newValue);
+        var response = await mediator.Send(req);
+        return response.ToResult();
```
Comment (rumkit) [inline CinemaCat.Api/Controllers/PersonsController.cs:36]: same here, let's change it to Created

File: CinemaCat.Api/Models/UploadedImagesInfo.cs
Comment (rumkit) [inline CinemaCat.Api/Models/UploadedImagesInfo.cs:1]: please move it to DTO
we consider models only those classes that represent collection items in DB

**Review by rumkit — CHANGES_REQUESTED** [review]

File: CinemaCat.Api/Controllers/ImageController.cs [hunk 4]
```diff
@@ -1,63 +1,36 @@
...
-
-            result.FullImageUrl = $"{Request.Scheme}://{Request.Host}{Url.RouteUrl("GetImage", new { id = guid })}";
-            result.CompressedImageUrl = $"{Request.Scheme}://{Request.Host}{Url.RouteUrl("GetImage", new { id = guid, compressed = true })}";
-        }
-
-        return Ok(result);
+        var req = new UploadImageRequest() { File = file };
+        var result = await mediator.Send(req);
+        var guid = result.Result.Id;
+        result.Result.FullImageUrl = $"{Request.Scheme}://{Request.Host}{Url.RouteUrl("GetImage", new { id = guid })}";
```
Comment (rumkit) [inline CinemaCat.Api/Controllers/ImageController.cs:32]: https://learn.microsoft.com/en-us/aspnet/core/mvc/controllers/routing?view=aspnetcore-8.0#generating-urls-by-action-name let's try this one and see if it results in a shorter code

**Review by rumkit — APPROVED** [review]
lgtm =)

File: CinemaCat.Api/Controllers/ImageController.cs (hunk 4 above)
Comment (mila-laurel) [reply CinemaCat.Api/Controllers/ImageController.cs:32]: done

**Pull request merged**